hazelcast/src/main/java/com/hazelcast/cache/impl/AbstractCacheService.java

```diff
@@ -588,10 +591,21 @@ private void deleteCacheResources(String name) {
     }
 
     @Override
+    public Operation getPreJoinOperation() {
+        Version clusterVersion = nodeEngine.getClusterService().getClusterVersion();
+        OnJoinCacheOperation preJoinCacheOperation = null;
+        if (clusterVersion.isGreaterOrEqual(V3_9)) {
```

Comment:
Added `assert !clusterVersion.isUnknown()` in [`AbstractCacheService.getPreJoinOperation()`](https://github.com/hazelcast/hazelcast/pull/10902/files#diff-f9947610f77bb138df37d08d6ade7f88R597)